src/util.ts

```diff
@@ -452,7 +462,97 @@ export function createMultiLinkFromLinks(links: string[]): string {
 		hashString += `&${key}=${value}`
 	}
 
-	return url.toString()
+	const shortenedLink = shortenMultilink(url.toString())
+
+	return shortenedLink
+}
+
+export function shortenMultilink(link: string): string {
+	const url = new URL(link)
+	const params = new URLSearchParams(url.search)
+
+	const i = params.get('i')
+	if (i) {
+		const numbers = i
+			.split(',')
+			.map((num) => parseInt(num, 10))
+			.sort((a, b) => a - b)
+
+		let grouped = []
+		let currentGroup = []
+		numbers.forEach((num) => {
+			if (currentGroup.length === 0) {
+				currentGroup.push(num)
+			} else if (num === currentGroup[currentGroup.length - 1] + 1) {
+				currentGroup.push(num)
+			} else {
+				grouped.push(`(${currentGroup[0]},${currentGroup.length})`)
+				currentGroup = [num]
+			}
+		})
+		if (currentGroup.length > 0) {
+			grouped.push(`(${currentGroup[0]},${currentGroup.length})`)
+		}
+
+		let queryString = ''
+		params.forEach((value, key) => {
+			if (key !== 'i') {
+				queryString += `${key}=${encodeURIComponent(value)}&`
+			}
+		})
+		queryString += `i=${grouped.join(',')}`
+
+		if (queryString.endsWith('&')) {
+			queryString = queryString.substring(0, queryString.length - 1)
+		}
+
+		const newUrl = url.origin + url.pathname + '?' + queryString + url.hash
+
+		return newUrl
+	} else {
+		throw new Error('Error shortening the multilink')
+	}
+}
+
+export function expandMultilink(link: string): string {
+	const url = new URL(link)
+	const params = new URLSearchParams(url.search)
+
+	let queryString = ''
+	params.forEach((value, key) => {
```

Comment:
Same comment about re-using the current `URLSearchParams` and `URL` instead of iterating through everything

Reply:
Updated the two functions to not overcomplicate things. Was having issues with encoding, but changed it that it just decodes after converting toString(). Thanks for suggesting